### PR TITLE
Allows for empty content type by default. Fixes #214

### LINKED ISSUE #214: Content-Type Transparency
Currently, if no `Content-Type` is present in the response header, fasthttp sets a default `Content-Type` to `text/plain; charset=utf-8`:  https://github.com/valyala/fasthttp/blob/ef299a95e2a898859c8652374b33cc7553b1532e/header.go#L270

We would like to provide a transparent proxy and do not want to add (or default) the `Content-Type`, if it is not present. Is there any way around this or would you be interested in a PR that adjusts this behavior?

### PR DESCRIPTION
**Pull request by davidbyttow**

### CODE REVIEW

**Comment by erikdubbelboer** [comment]
A minor problem I see is that `ResponseHeader.ContentType()` is a public API as well that is changing behavior now. There might be people who are using `ResponseHeader` directly in which case the behavior will change. But seeing as `ResponseHeader.ContentType()` never contained any documentation about it returning a default I guess it's not really breaking any contract.
Most people will be using the server which doesn't change with this. So I guess I approve.

**Comment by davidbyttow** [comment]
Yeah I thought about that too and reasoned the same thing. 

One alternative would be to change the logic in `ResponseHeader` to instead use `if contentType == nil` rather than `if len(contentType) == 0` and then add `DefaultContentType` to `Server` which defaults to nil and only calls `ResponseHeader.SetContentType` if it's non-nil.

**Comment by davidbyttow** [comment]
Here's an alternative, but while fewer changes, it may be a little bit more brittle/implicit. https://github.com/valyala/fasthttp/pull/407

**Comment by davidbyttow** [comment]
@erikdubbelboer should we merge this?

**Comment by erikdubbelboer** [comment]
@kirillDanshin what do you think?

**Comment by kirillDanshin** [comment]
I think we should

File: server.go [hunk 1]
```diff
@@ -1702,6 +1713,9 @@ func (s *Server) serveConn(c net.Conn) error {
 		if serverName != nil {
 			ctx.Response.Header.SetServerBytes(serverName)
 		}
+		if contentType != nil {
+			ctx.Response.Header.SetContentTypeBytes(contentType)
```
Comment (erikdubbelboer) [inline server.go]: Doesn't this always overwrite the header no matter if the user set it first or not?

Comment (davidbyttow) [reply server.go]: Well, isn't this called *before* the Handler? So it's just setting prior to user code.

**Comment by davidbyttow** [comment]
OK, I changed this by threading through the flag.

**Comment by davidbyttow** [comment]
FYI @erikdubbelboer ^

File: server_test.go [hunk 2]
```diff
@@ -1456,7 +1461,7 @@ func TestCompressHandler(t *testing.T) {
 	hh(&ctx)
 	s = ctx.Response.String()
 	br = bufio.NewReader(bytes.NewBufferString(s))
-	if err := resp.Read(br); err != nil {
+	if err = resp.Read(br); err != nil {
```
Comment (erikdubbelboer) [inline server_test.go]: Sorry if I'm being pedantic but please revert this change. It's always better to have variables scoped as tightly as possible.

File: server_test.go [hunk 3]
```diff
@@ -1479,7 +1484,7 @@ func TestCompressHandler(t *testing.T) {
 	h(&ctx)
 	s = ctx.Response.String()
 	br = bufio.NewReader(bytes.NewBufferString(s))
-	if err := resp.Read(br); err != nil {
+	if err = resp.Read(br); err != nil {
```
Comment (erikdubbelboer) [inline server_test.go]: Same as above.

File: server_test.go (hunk 2 above)
Comment (davidbyttow) [reply server_test.go]: The reason this was done is because it's actually a golinter warning about shadowing.

**Comment by davidbyttow** [comment]
Should be good to go @erikdubbelboer

File: header.go [hunk 4]
```diff
@@ -652,6 +653,7 @@ func (h *ResponseHeader) DisableNormalizing() {
 // Reset clears response header.
 func (h *ResponseHeader) Reset() {
 	h.disableNormalizing = false
+	h.noDefaultContentType = false
```
Comment (erikdubbelboer) [inline header.go]: I'm wondering now if this should be done here or in `resetSkipNormalize()`. I don't think it matters much but maybe doing it in `resetSkipNormalize()` is cleaner as everything is reset there.

**Comment by erikdubbelboer** [comment]
@kirillDanshin you have any thoughts about this or can I merge it?

**Comment by davidbyttow** [comment]
fyi @kirillDanshin ^

**Comment by kirillDanshin** [comment]
@erikdubbelboer @davidbyttow ok, let's merge it. sorry for the delay.
